backend/libs/watcher-operations.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { DocumentClient } from 'aws-sdk/clients/dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import { Room, VideoSyncStatus } from '../../extension/src/types';
+import {
+    MediaEventType,
+    PlayerEvent,
+} from './../../extension/src/contentscript/player';
+import { UserVideoStatus } from './../../extension/src/types';
+import { createRoom, joinExistingRoom, updateRoom } from './room-operations';
+import { updateWatcherVideoStatus } from './watcher-operations';
+
+const isTest = process.env.JEST_WORKER_ID;
+const config = {
+    convertEmptyValues: true,
+    ...(isTest && {
+        endpoint: 'localhost:8000',
+        sslEnabled: false,
+        region: 'local-env',
+    }),
+};
+const ddb = new DocumentClient(config);
+const tableName = 'watch-with-me-dev-RoomTable';
+
+let roomId: string;
+let room: Room | undefined = undefined;
+let playerEvent: PlayerEvent;
+let ddbRoom: DocumentClient.AttributeMap | undefined = undefined;
+
+const getDDBRoom = async () => {
+    const { Item: item } = await ddb
+        .get({
+            TableName: tableName,
+            Key: { roomId },
+        })
+        .promise();
+    if (!item) {
+        throw new Error('Error retrieving DDB room');
+    }
+    ddbRoom = item;
+};
+
+describe('#updateWatcher', () => {
+    beforeEach(async (done) => {
+        roomId = uuidv4();
+        const testRoom = await createRoom(roomId, tableName, 'owner', ddb);
+        if (!testRoom) {
```

Comment:
Is it possible ?

Reply:
no, but because of typescript, it's either this or I have to use some bangs everywhere. I'm actually not very consistent in the use of either guard clauses or bangs. I guess it's fine to always use bangs in tests since we more or less control the environment ?

Reply:
I decided to use bangs everywhere where it was needed. Makes test code look cleaner, and if it crashes... it crashes and it's fine